src/common/sharedlibrary/CMakeLists.txt

```diff
@@ -20,6 +20,11 @@ add_library(mirsharedsharedlibrary OBJECT
   shared_library_prober.cpp
 )
 
+# we overload dlvsym() for systems that don't have it
```

Comment:
I'd prefer if this were detected and `#ifdef`ed rather than silently providing a override.

[check_symbol_exists()](https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html#module:CheckSymbolExists) would appear to be the relevant CMake incantation.

Reply:
@RAOF please feel free to reactivate #683 and discuss that approach.

I dislike using #ifdef for alternative implementations as it doesn't even provide basic syntax checking of the alternative code.